article/2020/functional-programming-explained-in-python-javascript-and-java.md

```diff
@@ -203,60 +202,60 @@ def compose(*functions):
     return functools.reduce(compose2, functions, lambda x: x)
 ```
 
-At **line 4**, we’ll define a function `compose2` that takes two functions as arguments `f` and `g`.
-At **line 5**, we return a new function that represents the composition of `f` and `g`.
+在 **第 4 行**，我们定义了 `compose2` ，它接收两个函数 `f` 和 `g` 作为参数。
+在 **第 5 行**，我们返回了一个表示 `f` 和 `g` 的组合的新函数。
 
-Finally, at **line 6**, we return the results of our composition function.
+最后，在 **第 6 行**，我们返回了组合函数的结果。
 
-## Functional Programming in JavaScript
+## 函数式编程 —— 使用 JavaScript
 
-JavaScript has long offered functional capabilities due to its support for first-class functions. Functional programming has recently become more popular in JavaScript because it boosts performance when used in frameworks like Angular and React.
```

Comment:
- **first-class functions** 头等函数

- because it boosts performance when used in frameworks like Angular and React. 因为它可以像 Angular 和 React 一 
   样使用在框架中提升性能。